### PR TITLE
Add support for an initial directive argument without a name

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: 

## Summary

This adds support for the first directive argument to be unlabeled / unnamed. This can be useful—avoid duplication in the authoring syntax—when the directive name acts also describes the first argument. For example:
```
@DisplayName("A custom name")
```
instead of 
```
@DisplayName(name: "A custom name")
```

Only the first argument is can be unlabeled / unnamed. I feel that this provides most of the authoring syntax benefits without impacting the ability to diagnose directive argument syntax errors.

## Dependencies

n/a

## Testing

I will update this with testing instructions when I mark this as non-draft.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: Sources/Markdown/Markdown.docc/Markdown.md [hunk 1]
```diff
@@ -36,3 +36,5 @@ The markup tree provided by this package is comprised of immutable/persistent, t
 
 - <doc:VisitMarkup> 
 - <doc:FormatterAndOptions>
+
+<!-- Copyright (c) 2021 Apple Inc and the Swift Project authors. All Rights Reserved. -->
```
Comment (QuietMisdreavus) [inline Sources/Markdown/Markdown.docc/Markdown.md:40]: Should this one's copyright years be updated, even if the update that happened this year was the addition of the copyright line itself? 🤔 Probably fine either way.

**Comment by QuietMisdreavus** [comment]
I like this idea! The PR looks good so far.

@swift-ci Please test

**Comment by d-ronnqvist** [comment]
I've updated the documentation to reflex the syntax changes for name value arguments. This is ready for a review now.

**Review by QuietMisdreavus — APPROVED** [review]
Looks good, thanks!

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**